compression/gzip_zinfo.c

```diff
@@ -223,72 +295,55 @@ int generate_zinfo(FILE* in, offset_t span, struct gzip_zinfo** idx)
     (void)inflateEnd(&strm);
     free_zinfo(index);
     return ret;
-
 }
 
-int has_bits(struct gzip_zinfo* index, int checkpoint)
-{
-    if (checkpoint >= decode_int32(index->have))
-    {
-        return 0;
-    }
-    return index->list[checkpoint].bits != 0;
-}
-
-static uint8_t get_bits(struct gzip_zinfo* index, int checkpoint)
-{
-    return index->list[checkpoint].bits;
-}
-
-offset_t get_ucomp_off(struct gzip_zinfo* index, int checkpoint)
-{
-    return decode_offset(index->list[checkpoint].out);
-}
-
-offset_t get_comp_off(struct gzip_zinfo* index, int checkpoint)
-{
-    return decode_offset(index->list[checkpoint].in);
-}
-
-static int min(int lhs, int rhs)
-{
-    return lhs < rhs ? lhs : rhs;
+int generate_zinfo_from_file(const char *filepath, offset_t span, struct gzip_zinfo **index) {
+    FILE *fp = fopen(filepath, "rb");
+    if (fp == NULL)
+        return GZIP_ZINFO_FILE_NOT_FOUND;
+    int ret = generate_zinfo_from_fp(fp, span, index);
+    fclose(fp);
+    return ret;
 }
 
-// This is the same as extract_data_fp, but instead of a file, it decompresses data from a buffer which contains the exact data to decompress 
-int extract_data_from_buffer(void* d, offset_t datalen, struct gzip_zinfo* index, offset_t offset, void* buffer, offset_t len, int first_checkpoint)
-{
+int extract_data_from_fp(FILE *in, struct gzip_zinfo *index, offset_t offset, void *buffer, int len) {
```

Comment:
This function doesn't have change other than format and changing to the `init_flate` helper